src/coreclr/src/System.Private.CoreLib/ILLinkTrim.xml

```diff
@@ -42,14 +42,6 @@
       <!-- Methods is used by VS Tasks Window. -->
       <method name="GetActiveTaskFromId" />
     </type>
-    <type fullname="System.Collections.Generic.GenericArraySortHelper`1">
```

Comment:
Should we also remove these from the Mono ILLinkTrim file?

https://github.com/dotnet/runtime/blob/b463ce262ae303c6528d2876f36896af9fe3cb0e/src/mono/netcore/System.Private.CoreLib/src/LinkerDescriptor/System.Private.CoreLib.xml#L780-L787

Reply:
oh good point, I saw that this code lived in the System.PrivateCoreLib folder as opposed to the shared one, which made me assume this would just apply to Core, if that's not the case I can take care of it on the Mono side as well.

Reply:
Yeah, I'm a bit confused as well. Maybe it was just a copy-paste error to put it into the Mono xml file. I see that `ArraySortHelper.cs` is included in the shared projects:

https://github.com/dotnet/runtime/blob/6e515000fdd947b645984c2ac4103bf1a4681f0c/src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems#L133

But I don't see anywhere inside the Mono CoreLib that is using/calling it. So maybe it just can safely be removed from Mono CoreLib's ILLinkTrim file.

Reply:
The only thing that makes me hesitant whether or not to remove is that GenericArraySortHelper is actually part of mono's corelib, I do agree it is most likely not used because all it has is the constructor, but I wonder if there is anything else that might be using this class via reflection.

Reply:
Please remove it, it should not be used

Reply:
FYI these were copied from coreclr's ILLinkTrim.xml when we did the initial migration to dotnet/runtime, see the comment on L746 in Mono's ILLinkTrim.xml: https://github.com/dotnet/runtime/blob/f2786223508c0c70040fbf48ec3a39a607dd7f75/src/mono/netcore/System.Private.CoreLib/src/LinkerDescriptor/System.Private.CoreLib.xml#L746